scripts/lib/CIME/BuildTools/macrowriterbase.py

```diff
@@ -132,7 +132,7 @@ def write_line(self, line):
 
         A trailing newline is added, whether or not the input has one.
         """
-        self.output.write(str(self.indent_string() + line + "\n"))
+        self.output.write(u"{}{}\n".format(self.indent_string(), line))
```

Comment:
In cmakemacroswriter.py you went from u"{}" to str({}) here you are going the other way around - which should it be?

Reply:
Both py2 and py3 expect this value to be a unicode str. This wasn't caught due to doctests not being run.